web/frontend/src/components/base.vue

```diff
@@ -127,22 +133,33 @@ const digestInput = () => {
   req.setAngular(angular);
 
   rcLogConditionally(req);
-  props.client.baseService.setPower(req, new grpc.Metadata(), displayError);
-};
+  props.client.baseService.setPower(req, new grpc.Metadata(), async (error) => {
+    displayError(error);
+
+    if (pressed.size <= 0) {
+      await stop();
+    }
+
+    resolve();
+    console.log('INPUT STOP');
+  });
+});
 
-const handleKeyDown = (key: Keys) => {
+const handleKeyDown = async (key: Keys) => {
+  console.log(`pressed add ${key}`);
   pressed.add(key);
-  digestInput();
+  await digestInput();
```

Comment:
If we take this approach, then it looks like there's no reason to `await` these anymore, right?

Reply:
As it stands, it is still a `Promise` so while it may not be necessary I also don't think it hurts to be explicit it with it. But `stop` and `digestInputs` don't necessarily need to be `Promise`s anymore with this approach either. I intend to do another pass to take out the console logging so I can clean that up in the next pass.